Strings internationalization : french

arofarn opened the pull request:
First submission for review.

dglaude commented:
Suggestion:

> 481 #: py/compile.c:2271
> msgid "LHS of keyword arg must be an id"
> msgstr "La partie gauche d'un argument mot-clef doit être un identifiant"

I believe this is relative to "named argument" or "optional argument" so there might be a better french name for this than "argument mot-clef".

dglaude commented:
Suggestion:

> 562 #: py/emitnative.c:2126
> msgid "casting"
> msgstr "Conversion de type"
 
Personally I say "typage" but this might be "franglish".

dglaude commented:
Please merge those two lines 810+811:

> msgstr "on ne peut envoyé une valeur différente de None à un générateur"
> "fraîchement démarré"

dglaude commented:
Suggestion:

> 821 #: py/objgenerator.c:251
> msgid "can't pend throw to just-started generator"
> msgstr "on ne peut ???lever une exception en attente??? à/dans un générateur fraîchement démarré"
> 

This is very tricky. I believe "pend" is for "pending" and "throw" is for "throwing an exception".
Maybe you should ask an english rewording or explanation of the message to be sure to translate the right thing. I guess you don't get that kind of message frequently. :-)

dglaude commented:
Seems good to me.

Apparently the source POT file has multi-line that start with an empty line like this:

> msgid ""

In that case, not all of your translation start with:

> msgstr ""

I believe I did check them all, very quickly, but I have a general good feeling about this translation.
Maybe you should re-check all the 'step' and 'slice' to have a consistent use?
Or always say "pas 'step'" and "tranche 'slice'"???

Here are a few pending (not counting msgid "can't pend throw to just-started generator")

242 Maybe keep a reference to 'pull' like in other user of 'pull'

> #: ports/atmel-samd/common-hal/digitalio/DigitalInOut.c:168
> msgid "Cannot get pull while in output mode"
> msgstr "Ne peux être tirer en mode 'output'"

768 Alternative suggestion:

> #: py/objarray.c:426
> msgid "lhs and rhs should be compatible"
> msgstr "La partie gauche et droite doivent être compatible"

949 I believe you MUST put the %c in your translation

> #: py/objstr.c:998
> #, c-format
> msgid "unknown conversion specifier %c"
> msgstr "spécification de conversion inconnue"

1435 Peut-être remplacer nvm par "mémoire non volatile":

> #: shared-bindings/nvm/ByteArray.c:111 shared-bindings/nvm/ByteArray.c:141
> msgid "Unable to write to nvm."
> msgstr "Impossible d'écrire sur nvm"

arofarn commented:
> Formating error on the following lines?:
> 35 #: main.c:148 main.c:221
> 67 #: main.c:240
> 81 #: main.c:246
> 89 #: main.c:247
> 954 #: py/objstr.c:1029
> 1154 #: py/persistentcode.c:223
> 1320 #: shared-bindings/audiobusio/PDMIn.c:208
> 1326 #: shared-bindings/audioio/RawSample.c:98
> 1443 #: shared-bindings/pulseio/PWMOut.c:164
> 1449 #: shared-bindings/pulseio/PWMOut.c:195
> 1540 #: shared-bindings/util.c:38

??? I see nothing wrong

> Suggestion:
> 
>     821 #: py/objgenerator.c:251
>     msgid "can't pend throw to just-started generator"
>     msgstr "on ne peut ???lever une exception en attente??? à/dans un générateur fraîchement démarré"
> 
> This is very tricky. I believe "pend" is for "pending" and "throw" is for "throwing an exception".
> Maybe you should ask an english rewording or explanation of the message to be sure to translate the right thing. I guess you don't get that kind of message frequently. :-)

Very tricky indeed. This is why I left this one untranslated for now. I founded this : https://github.com/micropython/micropython/commit/63644016669c173190c71d39fb33ad9502fe2b0f

> Apparently the source POT file has multi-line that start with an empty line like this:
>     msgid ""
> 
> In that case, not all of your translation start with:
>     msgstr ""

I tried to keep the same number of lines for each string as often as possible. Sometimes I used the first blank line to avoid long lines. For example:
```#: main.c:240
msgid ""
"You are running in safe mode which means something really bad happened.\n"
msgstr "Vous êtes en mode sans-échec ce qui signifie que quelque chose de"
"mauvais est arrivé.\n"
```

About 'step' : it depends on context. sometime it mean "pas" or "étape" but in some cases it refers to the parameter. I corrected some of them and changed 'tranche' for 'slice' everywhere (french doc. use the english word 'slice')

dglaude commented:
I reviewed, but I do not have write access... someone with right access should do something to unblock.

Anton-2 commented on locale/fr.po:
```diff
@@ -0,0 +1,1599 @@
+
+#: shared-module/bitbangio/SPI.c:177
+msgid "Cannot read without MISO pin."
+msgstr "Ipossible de lire sans broche MISO."
```
Typo: Impossible

Anton-2 commented on locale/fr.po:
```diff
@@ -0,0 +1,1599 @@
+
+#: shared-bindings/busio/UART.c:98
+msgid "bits must be 7, 8 or 9"
+msgstr "les bits doivent être 7, 8 ou 9"
```
bits is an argument name, so maybe drop 'Les' : bits doit être 7, 8 ou 9

Anton-2 commented on locale/fr.po:
```diff
@@ -0,0 +1,1599 @@
+
+#: shared-bindings/busio/I2C.c:120
+msgid "Function requires lock."
+msgstr "La fonction nécessité un verrou."
```
Typo: La fonction nécessite un verrou.

Anton-2 commented on locale/fr.po:
```diff
@@ -0,0 +1,1599 @@
+#: shared-bindings/bitbangio/I2C.c:111 shared-bindings/bitbangio/SPI.c:121
+#: shared-bindings/busio/SPI.c:133
+msgid "Function requires lock"
+msgstr "La fonction nécessité un verrou"
```
Typo: La fonction nécessite un verrou.

Anton-2 commented on locale/fr.po:
```diff
@@ -0,0 +1,1599 @@
+#: py/runtime.c:1539
+#, c-format
+msgid "memory allocation failed, allocating %u bytes"
+msgstr "l'allocation de mémoire a échouée en allouant %u octects"
```
Typo : l'allocation de mémoire a échouée en allouant %u octets

arofarn replied:
- échouée
+ échoué ;)

Anton-2 replied:
Yes, échoué ;-)

Anton-2 commented on locale/fr.po:
```diff
@@ -0,0 +1,1599 @@
+
+#: py/runtime.c:1535
+msgid "memory allocation failed, heap is locked"
+msgstr "l'allocation de mémoire a échouée, la pile est vérrouillé"
```
Typo: l'allocation de mémoire a échoué, la pile est verrouillée

arofarn commented:
Last blocking thing that was missing was the charset. I built and uploaded a french firmware successfully on Feather M0 express and Feather M4 express boards.
It looks OK, even special character like "éèà" :)

Anton-2 commented on locale/fr.po:
```diff
@@ -0,0 +1,1599 @@
+
+#: py/objnamedtuple.c:92
+msgid "can't set attribute"
+msgstr "ne peut modifié un attribut"
```
typo: "ne peut modifier un attribut", but maybe better as "attribut non modifiable"

arofarn replied:
shorter & better

arofarn commented:
@dglaude : Ican only give you access to my fork =>  https://github.com/arofarn/circuitpython/tree/patch-1

Anton-2 commented on locale/fr.po:
```diff
@@ -0,0 +1,1599 @@
+
+#: py/objgenerator.c:108
+msgid "can't send non-None value to a just-started generator"
+msgstr "on ne peut envoyé une valeur différente de None à un générateur fraîchement démarré"
```
Typo : "on ne peut envoyer une valeur différente de None à un générateur fraîchement démarré"

Anton-2 commented on locale/fr.po:
```diff
@@ -0,0 +1,1599 @@
+
+#: py/obj.c:487
+msgid "object is not subscriptable"
+msgstr "l'objet n'est souscrivable"
```
Typo: "l'objet n'est pas souscrivable", and I'm not sure about souscrivable ...

arofarn replied:
"sous-scriptable" should be better

Anton-2 commented on locale/fr.po:
```diff
@@ -0,0 +1,1599 @@
+#: py/obj.c:490
+#, c-format
+msgid "'%s' object is not subscriptable"
+msgstr "l'objet '%s' n'est souscrivable"
```
Typo: "l'objet n'est pas souscrivable"

Anton-2 commented on locale/fr.po:
```diff
@@ -0,0 +1,1599 @@
+
+#: py/obj.c:89
+msgid "Traceback (most recent call last):\n"
+msgstr "Trace (appel les plus récents en dernier):\n"
```
Typo: Trace (appels les plus récents en dernier):\n

Anton-2 commented on locale/fr.po:
```diff
@@ -0,0 +1,1599 @@
+
+#: py/compile.c:2193
+msgid "super() can't find self"
+msgstr "super() ne peut pas trouvé self"
```
Typo: "super() ne peut pas trouver self"

Anton-2 commented on locale/fr.po:
```diff
@@ -0,0 +1,1599 @@
+
+#: py/compile.c:1188
+msgid "identifier redefined as nonlocal"
+msgstr "identifiant redéfini comme global"
```
Typo: "identifiant redéfini comme nonlocal"

Anton-2 commented on locale/fr.po:
```diff
@@ -0,0 +1,1599 @@
+
+#: py/compile.c:397 py/compile.c:542
+msgid "can't assign to expression"
+msgstr "ne peut pas assigné à l'expression"
```
Typo: "ne peut pas assigner à l'expression"

Anton-2 commented on locale/fr.po:
```diff
@@ -0,0 +1,1599 @@
+
+#: ports/atmel-samd/common-hal/busio/I2C.c:119
+msgid "Unsupported baudrate"
+msgstr "Débit en bauds non supporté"
```
Maybe just "Débit non supporté", there is no real notion of "bauds" here

arofarn replied:
I agree and it's shorter without losing its sense

Anton-2 commented on locale/fr.po:
```diff
@@ -0,0 +1,1599 @@
+#: ports/atmel-samd/common-hal/audiobusio/I2SOut.c:147
+#: ports/atmel-samd/common-hal/audiobusio/I2SOut.c:150
+msgid "Invalid bit clock pin"
+msgstr "Broche invalide pour le bit d'horloge"
```
'bit d'horloge' seems strange : it's the clock used to measure the time of one bit in the stream, so more like 'horloge des bits'. But it maybe easier to use 'bit clock', as it's the name of the signal.

arofarn replied:
I changed line #120 too for consistency:
```
msgstr "'bit clock' et 'word select' doivent partagé une horloge"
````

Anton-2 commented on locale/fr.po:
```diff
@@ -0,0 +1,1599 @@
+
+#: main.c:166
+msgid "Running in safe mode! Not running saved code.\n"
+msgstr "Mode sans-échec! Le code sauvegardé ne tourne pas.\n"
```
Maybe "ne tourne pas." -> "ne s'execute pas." ?

tannewt commented:
Thanks! Looks like a number of builds are running out of memory. I'm working on compressing them now so we should be able to check this in within the next few days.

arofarn commented:
I'm working on it. Lots of new strings ! 

I'm not sure how to translate 'advertisment' in the context of bluetooth : 'visibilité', 'publicité'... ?
There is some other (few) strings too cryptic for me or difficult to translate in french properly, but I guess.
It builds for me on feather M0 express, feather M4 express (I let travis checks for the rest)

tannewt commented:
No need to translate everything in the first PR! Anything helps.

Rebasing is needed though to ensure everything still builds. Let me know when that is done and then we can merge. Thanks!

arofarn commented:
Rebase done, but I can't build locally anymore. I got some errors in ../../shared-bindings/i2cslave/I2CSlave.c :

> error: passing argument 1 of 'mp_raise_ValueError' from incompatible pointer type [-Werror=incompatible-pointer-types]

It seems to expect compressed strings...
